src/OrchardCore.Themes/TheBlogTheme/Recipes/blog.recipe.json

```diff
@@ -1311,7 +1311,7 @@
             {
               "$type": "OrchardCore.AdminMenu.AdminNodes.LinkAdminNode, OrchardCore.AdminMenu",
               "LinkText": "Blog",
-              "LinkUrl": "[js: 'Admin/Contents/ContentItems/' +  variables('blogContentItemId') + '/Display']",
+              "LinkUrl": "[js: adminurlprefix() + '/Contents/ContentItems/' +  variables('blogContentItemId') + '/Display']",
```

Comment:
@Skrypt

We already have a `ConfigurationMethodProvider` so as i remember maybe we can already use something like this

    "[js:configuration('OrchardCore_Admin').AdminUrlPrefix]"

Reply:
We can't assume that someone will have provided a AdminUrlPrefix in a appsettings.json file. By exposing the AdminOptions class we then get the AdminUrlPrefix that defaults to `/admin`

Reply:
Else we would need to be able to use a ?? operator here.

Reply:
`[js: configuration('OrchardCore_Admin').AdminUrlPrefix ?? '/admin']`

Reply:
I'm getting an error when doing that

ParserException: Line 1: Unexpected token ?
Esprima.JavaScriptParser.ThrowUnexpectedToken<T>(Token token, string message)

Reply:
In the related issue i said

> Hmm, maybe we could change the `ConfigurationMethodProvider` to do a bind on an object instance, so that, if there is no config, we get the default value that is `Admin`. But not sure we can do that here as we would need a typed object

Maybe a null check in javascript but not so good

Reply:
Esprima.NET support only ECMAScript 2016 and the null coalescing operator is 2020.

Reply:
Yes, I think @jtkech is right we should use the `ConfigurationMethodProvider`.

But it has a few limitations, that I've known about for a while, but are easily fixed.

`"[js:configuration('OrchardCore_Admin').AdminUrlPrefix]"` doesn't work, because the object returned isn't accessible. it only works when you return a single string value...
i.e.
```
"[js:configuration('Foo')]"
```
where 'Foo' is
```
{
  "Foo": "/MyAdmin" 
}
```

I mentioned it recently here, https://github.com/OrchardCMS/OrchardCore/issues/7508 but time...

it's an easy fix (non breaking on how it works already), so I will change it to work better, i.e.
`"[js:configuration('OrchardCore_Admin:AdminUrlPrefix, 'Default')]"`

Reply:
Don't waste time on this. We need to remove this code. It's not secure.